### PR TITLE
Some fixes for the test script template

### PR DESCRIPTION
**Pull request by vbehar**
A few fixes for the "test" script template : 
- support for running on OSX
- fix incremental build
- more logs

### CODE REVIEW

File: pkg/create/templates/test.go [hunk 1]
```diff
@@ -106,7 +114,7 @@ test_connection() {
   local result=1
   while [ $attempt -le $max_attempts ]; do
     echo "Sending GET request to http://$(container_ip):${test_port}/"
-    response_code=$(curl -s -w %{http_code} -o /dev/null http://$(container_ip):${test_port}/)
+    response_code=$(docker run --rm -it sequenceiq/alpine-curl curl -s -w %{http_code} -o /dev/null http://$(container_ip):${test_port}/)
```
Comment (bparees) [inline pkg/create/templates/test.go]: this seems a little heavy... is curl not available on osx?

Comment (vbehar) [inline pkg/create/templates/test.go]: yes it is, but on osx if using boot2docker, the container IP is not directly accessible. thus the use of an container, to be in the same network.

Comment (vbehar) [inline pkg/create/templates/test.go]: if you want, I can add a check to ping the IP for example, and use the container to curl only if it is not accessible

Comment (bparees) [inline pkg/create/templates/test.go]: I see.  no i think you can leave it, but leave the original line there, commented out, with a comment about why it's being done that way, so people who don't care can easily revert the behavior.

Comment (vbehar) [inline pkg/create/templates/test.go]: ok

File: pkg/create/templates/test.go [hunk 2]
```diff
@@ -106,7 +114,9 @@ test_connection() {
   local result=1
   while [ $attempt -le $max_attempts ]; do
     echo "Sending GET request to http://$(container_ip):${test_port}/"
-    response_code=$(curl -s -w %{http_code} -o /dev/null http://$(container_ip):${test_port}/)
+    # here, we run the curl inside a container for people running on boot2docker (who can't access the container's IP directly)
+    #response_code=$(curl -s -w %{http_code} -o /dev/null http://$(container_ip):${test_port}/)
+    response_code=$(docker run --rm -it sequenceiq/alpine-curl curl -s -w %{http_code} -o /dev/null http://$(container_ip):${test_port}/)
```
Comment (soltysh) [inline pkg/create/templates/test.go]: I'm a bit worried this introduces yet another dependency `sequenceiq/alpine-curl` which is not desired in tests, imho.

Comment (vbehar) [inline pkg/create/templates/test.go]: I understand (even though it's just a template, so people are free to change it before running it).
what about reverting to the original behaviour, and just displaying a warning to osx users suggesting them to run the curl inside a container if they can't access the container's ip directly ?

Comment (soltysh) [inline pkg/create/templates/test.go]: That will be perfect!

**Comment by soltysh** [comment]
Btw. please squash all your commits into one and after you fix that last issue I'll merge this as is.

**Comment by vbehar** [comment]
ok, fixed and squashed

**Comment by soltysh** [comment]
@vbehar thanks for doing this. LGTM
[merge]

**Pull request merged**